aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -461,7 +489,9 @@ Tensor &all_out(Tensor &result, const Tensor &self, int64_t dim, bool keepdim) {
   if (_dimreduce_return_trivial(result, self, 1, dim, keepdim)) {
     return result;
   } else {
-    return at::legacy::th::_th_all_out(result, self, dim, keepdim);
+    ScalarType dtype = get_dtype(result, self, {}, true);
```

Comment:
`get_dtype` could make sense here but I don't think you want promote_integers to be true.